### ISSUE TITLE
Bug with lazy_load for evaluation script 

### ISSUE THREAD

**Issue by timothylimyl** [Lightning-AI/lit-llama#264]
Hi, when running `evaluate_lora.py` there will be an error that pops out:

Under debug, this is in the function `llama_model_lookup()`
```
'lazy_load' object is not subscriptable
```

But if you convert the evaluation script to how you load in the `generate_.py` script:

```
    with (lazy_load(pretrained_path) as pretrained_checkpoint,
          lazy_load(lora_path) as adapter_checkpoint):
        name = llama_model_lookup(pretrained_checkpoint)

        with EmptyInitOnDevice(
                device=fabric.device, dtype=dtype, quantization_mode=quantize
        ), lora(r=lora_r, alpha=lora_alpha, dropout=lora_dropout, enabled=True):
            model = LLaMA.from_name(name)

            # 1. Load the pretrained weights
            model.load_state_dict(pretrained_checkpoint, strict=False)
            # 2. Load the fine-tuned adapter weights
            model.load_state_dict(adapter_checkpoint, strict=False)
```

then the error goes away and you are able to load the model, I reckon you need the context manager for lazy_load work together with `llama_model_lookup`.

**Comment by carmocca** [comment]
I had the similar issues on Lit-Parrot on Windows. I had to unroll the context managers: https://github.com/Lightning-AI/lit-parrot/pull/3

Maybe that helps here

**Comment by carmocca** [comment]
Fixed by #299
